File: Shkolniy_etap_VSOSH_9_11_klass_3grupa_regionov/patties.cpp
#include <iostream>

long long int calculateNextBuy(long long int todaysPurchase)
{
	if (todaysPurchase == 1)
		return 2;
	else if (todaysPurchase == 2)
		return 3;
	else if (todaysPurchase == 3)
		return 1;
}

int main()
{
	using namespace std;
	long long int cranberriesPrice;
	long long int blueberriesPrice;
	long long int cherriesPrice;
	long long int days;
	long long int buyToday;
	long long int totalSpent = 0;
	cin >> cranberriesPrice;
	cin >> blueberriesPrice;
	cin >> cherriesPrice;
	cin >> days;
	cin >> buyToday;

	if (buyToday == 1)
		totalSpent += cranberriesPrice;
	else if (buyToday == 2)
		totalSpent += blueberriesPrice;
	else if (buyToday == 3)
		totalSpent += cherriesPrice;
	for (int i = 1; i < days; i++)
	{
		if (calculateNextBuy(buyToday) == 1)
			totalSpent += cranberriesPrice;
		if (calculateNextBuy(buyToday) == 2)
			totalSpent += blueberriesPrice;
		if (calculateNextBuy(buyToday) == 3)
			totalSpent += cherriesPrice;
		buyToday = calculateNextBuy(buyToday);
	}
	cout << totalSpent;
}

File: Shkolniy_etap_VSOSH_9_11_klass_3grupa_regionov/patties.py
def calculate_next_buy(todays_purchase):
    if todays_purchase == 1:
        return 2
    elif todays_purchase == 2:
        return 3
    elif todays_purchase == 3:
        return 1


def main():
    cranberries_price = int(input())
    blueberries_price = int(input())
    cherries_price = int(input())
    days = int(input())
    buy_today = int(input())
    total_spent = 0
    if(buy_today == 1):
        total_spent += cranberries_price
    if(buy_today == 2):
        total_spent += blueberries_price
    if(buy_today == 3):
        total_spent += cherries_price
    for i in range(1, days):
        if(calculate_next_buy(buy_today) == 1):
            total_spent += cranberries_price
        elif(calculate_next_buy(buy_today) == 2):
            total_spent += blueberries_price
        elif(calculate_next_buy(buy_today) == 3):
            total_spent += cherries_price
        buy_today = calculate_next_buy(buy_today)
    print(total_spent)


main()
